firebase/public/main.js

```diff
@@ -387,6 +389,31 @@ function displayVersions(originId) {
     document.querySelector('#lsb-version').innerHTML = message.lsb;
     document.querySelector('#sys-version').innerHTML = message.uname;
     document.querySelector('#daq-versions').classList.add('valid');
+    if (message.timestamp && message.timestamp < heartbeatTimestamp) {
```

Comment:
this code doesn't seem related to the encapsulating function (displayVersions)... I'd recommend extracting a bunch of this to a separate function, and then maybe adjusting the names to match what their role is?